spec/controllers/products_controller_spec.rb

```diff
@@ -7,68 +7,131 @@
   let(:tea) { create(:product, category: tea_category) }
   let(:error_message) { 'Product not found' }
 
-  let(:service_result) { instance_double(GetProductService, result: tea, success?: true) }
-  let(:service_failure) do
+  let(:product_result) { instance_double(GetProductService, success?: true, result: tea) }
```

Comment:
this should be named service, as it return result for service, not return result for product